internal/transport/controlbuf.go

```diff
@@ -190,10 +190,10 @@ type incomingGoAway struct {
 func (*incomingGoAway) isTransportResponseFrame() bool { return false }
 
 type goAway struct {
-	code      http2.ErrCode
-	debugData []byte
-	headsUp   bool
-	closeConn error // if set, loopyWriter will exit, resulting in conn closure
+	code         http2.ErrCode
+	debugData    []byte
+	headsUp      bool
+	closeConnErr error // if set, loopyWriter will exit with this error
```

Comment:
Still this is only true on the server side. Is this true on the client side as well?

Reply:
Yeah this is valid for client side as well. We are looping out from writer when we receive this error.